Services/AssessmentQuestion/interfaces/interface.ilAsqQuestion.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/* Copyright (c) 1998-2013 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+ * Interface ilAsqQuestion
+ *
+ * @author    Björn Heyser <info@bjoernheyser.de>
+ * @version    $Id$
+ *
+ * @package    Services/AssessmentQuestion
+ */
+interface ilAsqQuestion
+{
+	/**
+	 * @param int $parentId
+	 * @return void
+	 */
+	public function setParentId($parentId);
+	
+	/**
+	 * @return int
+	 */
+	public function getParentId();
+	
+	/**
+	 * @param int $questionId
+	 * @return void
+	 */
+	public function setId($questionId);
+	
+	/**
+	 * @return int $questionId
+	 */
+	public function getId();
+	
+	/**
+	 * @return string
+	 */
+	public function getTitle();
+	
+	/**
+	 * @return string
+	 */
+	public function getComment();
+	
+	/**
+	 * @return string
+	 */
+	public function getOwner();
+	
+	/**
+	 * @return string $questionType
+	 */
+	public function getQuestionType();
+	
+	/**
+	 * @return string
+	 */
+	public function getQuestionText();
+	
+	/**
+	 * @return string
+	 */
+	public function getPoints();
+	
+	/**
+	 * @return string
+	 */
+	public function getEstimatedWorkingTime();
+	
+	/**
+	 * @return void
+	 */
+	public function load();
+	
+	/**
+	 * @return void
+	 */
+	public function save();
+	
+	/**
+	 * @return void
+	 */
+	public function delete();
```

Comment:
You may want to consider the repository pattern for the persistance of questions, i.e. move the storage and loading into a seperate class. This might make it easier to fulfill you database decoupling goal, as well as unit test etc.

Reply:
yes you're right, this is a quite good idea. but i guess we agree each other when i say, this kind pattern is to done behind the interfaces, right?